app/controllers/houseConsignment/index/departureMeansOfTransport/CountryController.scala

```diff
@@ -0,0 +1,100 @@
+import navigation.DepartureTransportMeansNavigator
+import pages.houseConsignment.index.departureMeansOfTransport.CountryPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import repositories.SessionRepository
+import services.ReferenceDataService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewModels.houseConsignment.index.departureMeansOfTransport.HouseConsignmentCountryViewModel.HouseConsignmentCountryViewModelProvider
+import views.html.houseConsignment.index.departureMeansOfTransport.CountryView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CountryController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  actions: Actions,
+  formProvider: SelectableFormProvider,
+  referenceDataService: ReferenceDataService,
+  val controllerComponents: MessagesControllerComponents,
+  view: CountryView,
+  navigator: DepartureTransportMeansNavigator,
+  countryViewModelProvider: HouseConsignmentCountryViewModelProvider
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val prefix = "houseConsignment.index.departureMeansOfTransport.country"
+
+  def onPageLoad(arrivalId: ArrivalId, houseConsignmentIndex: Index, transportMeansIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId).async {
+      implicit request =>
+        referenceDataService
+          .getCountries()
+          .map(
+            country => SelectableList(country.toSeq)
+          ) map {
+          countries =>
+            val viewModel = countryViewModelProvider.apply(mode, houseConsignmentIndex)
+            def form(houseConsignmentIndex: Index, transportMeansIndex: Index) =
+              formProvider(mode, prefix, countries, houseConsignmentIndex.display, transportMeansIndex.display)
+            val preparedForm = request.userAnswers.get(CountryPage(houseConsignmentIndex, transportMeansIndex)) match {
+              case None        => form(houseConsignmentIndex, transportMeansIndex)
+              case Some(value) => form(houseConsignmentIndex, transportMeansIndex).fill(value)
+            }
+            Ok(view(preparedForm, countries.values, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, transportMeansIndex, mode, viewModel))
+        }
+    }
+
+  def onSubmit(arrivalId: ArrivalId, houseConsignmentIndex: Index, transportMeansIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId).async {
+      implicit request =>
+        referenceDataService
+          .getCountries()
+          .map(
+            country => SelectableList(country.toSeq)
+          ) flatMap {
+          countries =>
+            val viewModel = countryViewModelProvider.apply(mode, houseConsignmentIndex)
+            def form      = formProvider(mode, prefix, countries, houseConsignmentIndex.display)
```

Comment:
Is there any reason this form definition is different from the one in the onPageLoad method?

Reply:
No - I have updated it